lib/ansible/modules/files/jsonmod.py

```diff
@@ -0,0 +1,382 @@
+#!/usr/bin/python -tt
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Matthew Gamble <git@matthewgamble.net>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: jsonmod
+short_description: Update part of a JSON file without worrying about how it is formatted
+description:
+    - Used to update part of a JSON file by specifying a path to the node
+      you want to update and what you want to add to or replace it with.
+      It works regardless of how the JSON file is formatted.
+version_added: "1.0"
```

Comment:
This needs to be `2.5` or `2.6` depending on when the changes come in for the merge window. If you can get the changes in today, we should be good for the `2.5` merge window (apologies for the late review).

Reply:
I can get it done in about 12 hours time, is that okay?